Sources/InteroperabilityTests/TestService.swift

```diff
@@ -189,3 +207,38 @@ internal struct TestService: Grpc_Testing_TestService.ServiceProtocol {
     throw RPCError(code: .unimplemented, message: "The RPC is not implemented.")
   }
 }
+
+extension ServerRequest.Single {
+  fileprivate func extractInitialAndTrailingMetadata() -> (Metadata, Metadata) {
+    var initialMetadata = Metadata()
+    var trailingMetadata = Metadata()
+
+    for value in self.metadata[stringValues: "x-grpc-test-echo-initial"] {
+      initialMetadata.addString(value, forKey: "x-grpc-test-echo-initial")
+    }
+
+    for value in self.metadata[binaryValues: "x-grpc-test-echo-trailing-bin"] {
+      trailingMetadata.addBinary(value, forKey: "x-grpc-test-echo-trailing-bin")
+    }
+
+    return (initialMetadata, trailingMetadata)
+  }
+}
+
+@available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
+extension ServerRequest.Stream {
+  fileprivate func extractInitialAndTrailingMetadata() -> (Metadata, Metadata) {
+    var initialMetadata = Metadata()
+    var trailingMetadata = Metadata()
+
+    for value in self.metadata[stringValues: "x-grpc-test-echo-initial"] {
+      initialMetadata.addString(value, forKey: "x-grpc-test-echo-initial")
+    }
+
+    for value in self.metadata[binaryValues: "x-grpc-test-echo-trailing-bin"] {
+      trailingMetadata.addBinary(value, forKey: "x-grpc-test-echo-trailing-bin")
+    }
+
+    return (initialMetadata, trailingMetadata)
+  }
```

Comment:
There's a lot of needless repetition here. Instead can we have a `fileprivate` `extension` on `Metadata` which makes the initial and trailing metadata?

Then in the test functions we can just call `request.metadata.makeInitialAndTrailingMetadata()`?